python/lsst/daf/butler/direct_butler/_direct_butler.py

```diff
@@ -1661,29 +1678,39 @@ def _ingest_file_datasets(
                 groupedDataIds[group_key][ref.dataId] = dataset
             refs.extend(dataset.refs)
 
-        # Bulk insert.
+        # Ensure that dataset types are created and all ref information
+        # extracted.
         import_info = self._prepare_for_import_refs(
             self,
             refs,
             skip_missing=False,
             register_dataset_types=True,
             dry_run=dry_run,
-            transfer_dimensions=False,
+            transfer_dimensions=transfer_dimensions,
         )
+        return import_info
 
-        self._import_dimension_records(import_info.dimension_records, dry_run=dry_run)
-        imported_refs = self._import_grouped_refs(
-            import_info.grouped_refs, None, progress, dry_run=dry_run, expand_refs=True
-        )
+    def _ingest_file_datasets(
+        self,
+        datasets: Sequence[FileDataset],
+        import_info: _ImportDatasetsInfo,
+        progress: Progress,
+        *,
+        dry_run: bool = False,
+    ) -> None:
+        with self.transaction():
```

Comment:
`ingest_zip()` already has a transaction open when calling this method, do we want a nested transaction here?

Reply:
Good spot. That transaction is irrelevant.